sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -657,6 +659,18 @@ class Analyzer(
     }
   }
 
+  /**
+   * Resolve table relations with concrete relations from v2 catalog.
+   */
+  case class ResolveTables(lookupCatalog: LookupCatalog) extends Rule[LogicalPlan] {
```

Comment:
where do we use `lookupCatalog` in this rule?

Reply:
`CatalogObjectIdentifier` is defined in `LookupCatalog`. That matches the table identifier if it starts with a known catalog.

Reply:
Good catch. This parameter is no longer needed after moving the rule here since Analyzer has LookupCatalog mixin.

Reply:
Oh, right. Good one.